github_issue_summarization/ks-kubeflow/components/tensor2tensor.libsonnet

```diff
@@ -1,95 +1,148 @@
 {
-  parts(params):: {
-    apiVersion: "kubeflow.org/v1alpha1",
-    kind: "TFJob",
-    metadata: {
-      name: "tensor2tensor",
-      namespace: params.namespace,
-    },
-    spec: {
-      replicaSpecs: [
-        {
-          replicas: 1,
-          template: {
-            spec: {
-              containers: [
-                {
-                  image: params.image,
-                  name: "tensorflow",
-                  command: [
-                    "bash",
-                  ],
-                  args: [
-                    "/home/jovyan/train_dist_launcher.sh",
-                    "1",
-                    params.workers,
-                    "0",
-                    params.train_steps,
-                    "/job:master",
-                    "False",
-                  ],
-                },
-              ],
-              restartPolicy: "OnFailure",
+  parts(params, env):: {
+    // Define some defaults.
+    local updatedParams = {
+      sync: "0",
+      
+      dataDir: "gs://kubeflow-examples-data/gh_issue_summarization/data",
+      usrDir: "./github",
+      problem: "github_issue_summarization_problem",
+
+      model: "transformer_encoder",
+      hparams: "transformer_github_issues",
+      hparamsSet: "transformer_github_issues",      
+      outputGCSPath: "gs://kubecon-gh-demo/gh-t2t-out/temp",
+
+      gpuImage: null,
+      cpuImage: null,
+
+      trainSteps: 20000,
+      evalSteps: 10,
+
+      psGpu: 0,
+      workerGpu: 0,
+
+      workers: 3,
+      masters: 1,
+      ps: 1,
+
+      jobName: "tensor2tensor",
+    } + params,
+
+    local containerEnv = [
+      {
+        name: "PYTHONPATH",
+        value: "/home/jovyan",
+      }
+    ],
+    local baseCommand = [      
+      "/home/jovyan/github/t2t_launcher.sh",
+      "--train_steps=" + std.toString(updatedParams.trainSteps),
```

Comment:
nit: `std.toString` is redundant and can be removed from all places.

> String concatenation will implicitly convert one of the values to a string if necessary

http://jsonnet.org/ref/spec.html